(fix) Restore offline patient banner

icrc-jofrancisco opened the pull request:
## Requirements
- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.

#### For changes to apps
- [x]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and **design documentation**.

#### If applicable
- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary
<!-- Please describe what problems your PR addresses. -->
This pull request aims to restore the offline banner feature, ensuring that an appropriate banner is displayed when the application is offline. The implementation involves reintegrating the offline banner component and associated logic to enhance user experience during periods of network unavailability.

## Screenshots
<!-- Required if you are making UI changes. -->
Before:
![image](https://github.com/openmrs/openmrs-esm-core/assets/94977371/a05299c6-4103-45de-9389-319072d4a666)

After:
![Pasted Graphic 3](https://github.com/openmrs/openmrs-esm-core/assets/94977371/d253cf31-c777-472f-8c08-eff0c4462dbe)

## Related Issue
<!-- Paste the link to the Jira ticket here if one exists. -->
<!-- https://issues.openmrs.org/browse/O3- -->

## Other
<!-- Anything not covered above -->

Thanks,
CC: @ibacher

ibacher reviewed (commented):
Two things: 

1. I think it's better to keep to the convention we've established of `offlineBannerComponent`
2. It's preferable to have the exports in roughly the same order they are in the routes.json. In other words, the offlineBanner export should come after the localeChanger and before the linkComponent

icrc-jofrancisco commented:
> Two things:
> 
> 1. I think it's better to keep to the convention we've established of `offlineBannerComponent`
> 2. It's preferable to have the exports in roughly the same order they are in the routes.json. In other words, the offlineBanner export should come after the localeChanger and before the linkComponent

Thanks for the review @ibacher ! I've addressed the feedback and pushed the changes.

ibacher commented on packages/apps/esm-primary-navigation-app/src/index.ts:
```diff
@@ -48,6 +49,8 @@ export const userPanel = getSyncLifecycle(userPanelComponent, options);
 
 export const localeChanger = getSyncLifecycle(localeChangerComponent, options);
 
+export const offlineBannerComponent = getSyncLifecycle(OfflineBanner, options);
```
I meant rename the import, so `import offlineBannerComponent from ...` rather than export...

ibacher commented on packages/apps/esm-primary-navigation-app/src/index.ts:
```diff
@@ -13,7 +13,7 @@ import { syncUserLanguagePreference } from "./offline";
 import userPanelComponent from "./components/user-panel-switcher-item/user-panel-switcher.component";
 import localeChangerComponent from "./components/choose-locale/change-locale.component";
-import OfflineBanner from "./components/offline-banner/offline-banner.component";
+import OfflineBannerComponent from "./components/offline-banner/offline-banner.component";
```
```suggestion
import offlineBannerComponent from "./components/offline-banner/offline-banner.component";
```

ibacher commented on packages/apps/esm-primary-navigation-app/src/index.ts:
```diff
@@ -49,7 +49,7 @@ export const userPanel = getSyncLifecycle(userPanelComponent, options);
 export const localeChanger = getSyncLifecycle(localeChangerComponent, options);
 
-export const offlineBannerComponent = getSyncLifecycle(OfflineBanner, options);
+export const offlineBanner = getSyncLifecycle(OfflineBannerComponent, options);
```
```suggestion
export const offlineBanner = getSyncLifecycle(offlineBannerComponent, options);
```

ibacher commented:
Whoa! This went too far... I just needed things in the index.ts file to be named consistently. Don't touch the other files!

ibacher commented on packages/apps/esm-primary-navigation-app/src/components/navbar/navbar.component.tsx:
```diff
@@ -22,7 +22,7 @@ import { isDesktop } from "../../utils";
 import Logo from "../logo/logo.component";
 import NotificationsMenuPanel from "../navbar-header-panels/notifications-menu-panel.component";
-import OfflineBanner from "../offline-banner/offline-banner.component";
+import OfflineBannerComponent from "../offline-banner/offline-banner.component";
```
These changes are unnecessary! We just needed things in `index.ts` to be consistent. We don't need to rename it everywhere!
